### PR TITLE
Migration: associate ingredient records with product names instead of products

### PR DESCRIPTION
**Pull request by jayaddison**
### Describe the reason for these changes and the problem that they solve
Consider two recipes, that both involve the ingredient typically referred to as `cilantro` or `coriander` depending on locality.

The following ingredient description lines could appear in each of those recipes:

- A sprinkling of chopped `cilantro`
- Add a dash of `coriander`

We want to use our updated representation of synonyms (#54) to link both of these two references to a single `product` (that may have an opaque, non-human-meaningful ID).

In our current database schema. each of these lines is stored in the database as a [`RecipeIngredient`](https://github.com/openculinary/backend/blob/c1ceaa620382c17d0363f15502e9e4f9a18b514c/reciperadar/models/recipes/ingredient.py#L6-L76) -- in particular using the [`product_id` foreign key column](https://github.com/openculinary/backend/blob/c1ceaa620382c17d0363f15502e9e4f9a18b514c/reciperadar/models/recipes/ingredient.py#L12-L15).

We'd like to adjust the schema so that each of the ingredient lines notionally references a product _name_.

In other words: the two ingredient lines in the example above refer to the same underlying product, but by using two different `product_name` records.

```
A sprinkling of chopped <name>cilantro</name>   ------> name:cilantro  --\
                                                                          +-----> product:<opaque-id>
Add a dash of <name>coriander</name>            ------> name:coriander --/
```

### Briefly summarize the changes
1. Add a `product_name` foreign key column to the `RecipeIngredient` model
1. Proxy requests for `RecipeIngredient.product` via `ProductName` to the `Product` table

### How have the changes been tested?
1. Unit tests continue to pass

**List any issues that this change relates to**
Relates to #54.

### CODE REVIEW

**Comment by jayaddison** [comment]
> Unit tests continue to pass

(note: unit test coverage does include construction of `Recipe` objects and accesses to the `RecipeIngredient.product` field)
